dipy/nn/finta_model.py

```diff
@@ -0,0 +1,435 @@
+        space where the data will be encoded
+        Default: 32
+    kernel_size : int, optional
+        Length of the 1D kernel used in the
+        convolutional layers of the Encoder
+        and the Decoder
+        Default: 3
+
+    Returns
+    -------
+    keras.Model
+        Encoder model
+    keras.Model
+        Decoder model
+    tuple
+        Size of the convolutional output of
+        the Encoder block.
+    """
+    input_data = keras.Input(shape=(256, 3), name='input_streamline')
+
+    # encode
+    encoder = Encoder(latent_space_dims=latent_space_dims,
+                      kernel_size=kernel_size)
+    encoded = encoder(input_data)
+    # Instantiate encoder model
+    model_encoder = Model(input_data, encoded, name="Encoder")
+
+    # decode
+    decoder = Decoder(encoder.encoder_out_size,
+                      kernel_size=kernel_size)
+    decoded = decoder(encoded)
+    output_data = decoded
+    # Instantiate decoder model
+    model_decoder = Model(encoded, decoded, name="Decoder")
+
+    # Instantiate model and name it
+    return model_encoder, model_decoder, encoder.encoder_out_size
+
+
+class IncrFeatStridedConvFCUpsampReflectPadAE(Model):
+    """
+    Strided convolution-upsampling-based
+    AutoEncoder using reflection-padding and
+    increasing feature maps in decoder.
+    Architecture based on Jon Haitz Legarreta Gorroño's
+    PyTorch implementation, from his FINTA [1] work.
+    [1] https://doi.org/10.1016/j.media.2021.102126
+    """
+
+    def __init__(self, latent_space_dims=32, kernel_size=3,
+                 **kwargs):
+        super(IncrFeatStridedConvFCUpsampReflectPadAE, self).__init__(**kwargs)
+
+        # Parameter Initialization
+        self.kernel_size = kernel_size
+        self.latent_space_dims = latent_space_dims
+
+        model = init_model(latent_space_dims=self.latent_space_dims,
+                           kernel_size=self.kernel_size)
+        self.encoder = model[0]
+        self.decoder = model[1]
+        self.encoder_out_size = model[2]
+
+        # Model name is: "incr_feat_strided_conv_fc_upsamp_reflect_pad_ae"
+
+    def get_config(self):
+        base_config = super().get_config()
+        config = {
+            "latent_space_dims": keras.saving.serialize_keras_object(self.latent_space_dims),
+            "kernel_size": keras.saving.serialize_keras_object(self.kernel_size),
+            "encoder_out_size": keras.saving.serialize_keras_object(self.encoder_out_size),
+        }
+        return {**base_config, **config}
+
+    @classmethod
+    def from_config(cls, config):
+        latent_space_dims = keras.saving.deserialize_keras_object(config.pop('latent_space_dims'))
+        encoder_out_size = keras.saving.deserialize_keras_object(config.pop('encoder_out_size'))
+        kernel_size = keras.saving.deserialize_keras_object(config.pop('kernel_size'))
+        return cls(latent_space_dims, kernel_size, **config)
+
+    def call(self, x):
+        r"""
```

Comment:
Do not use raw strings.

Reply:
Got it, I was using them because I had seen them in other docstrings.